addons/resource/views/resource_calendar_views.xml

```diff
@@ -59,39 +53,125 @@
                     </div>
                     <group name="resource_details" col="2">
                         <group name="resource_working_hours">
-                            <field name="flexible_hours"/>
-                            <label for="full_time_required_hours" invisible="flexible_hours"/>
-                            <label for="full_time_required_hours" string="Hours per Week" invisible="not flexible_hours"/>
+                            <field name="schedule_type" widget="radio"/>
+                            <field name="flexible_hours" invisible="1"/>
+                            <label for="full_time_required_hours" invisible="not company_id"/>
+                            <label for="full_time_required_hours" string="Required Full Time" invisible="company_id"/>
                             <div>
-                                <field name="full_time_required_hours" style="width: 6ch;" widget="float_time"/>
+                                <field name="full_time_required_hours" style="width: 6ch;" widget="float_time" readonly="company_id"/>
                                 <span>hours/week</span>
                             </div>
-                            <field name="hours_per_day" widget="float_time" readonly="not flexible_hours"/>
-                            <label for="work_time_rate" string="Work Time Rate" invisible="flexible_hours"/>
-                            <div invisible="flexible_hours">
+                            <label for="work_time_rate" string="Work Time Rate" invisible="schedule_type == 'fully_flexible'"/>
+                            <div invisible="schedule_type == 'fully_flexible'">
                                 <field name="work_time_rate" nolabel="1"  style="width: 5ch;"/>
                                 <span class="ms-1">%</span>
                             </div>
-                            <field name="is_fulltime" invisible="1"/>
                         </group>
-                        <group name="resource_company">
-                            <field name="active" invisible="1"/>
-                            <field name="company_id" groups="base.group_multi_company" placeholder="Visible to all"/>
-                            <field name="tz" widget="timezone_mismatch" options="{'tz_offset_field': 'tz_offset', 'mismatch_title': 'Timezone Mismatch : This timezone is different from that of your browser.\nPlease, be mindful of this when setting the working hours or the time off.'}" />
-                            <field name="tz_offset" invisible="1"/>
+                        <group col="1">
+                            <group name="resource_company">
```

Comment:
Don't you find it strange to have the checkbox before the label ? Not very odoo style compared to other form views along the codebase.